packages/angular/src/builders/module-federation-dev-server/module-federation-dev-server.impl.ts

```diff
@@ -107,25 +115,71 @@ export function executeModuleFederationDevServerBuilder(
       }
     }
 
-    scheduleTarget(
-      context.workspaceRoot,
+    let outWithErr: null | string[] = [];
+    const remoteProcess = fork(
+      nxBin,
+      [
+        'run',
+        `${remote}:${target}${
+          context.target.configuration ? `:${context.target.configuration}` : ''
+        }`,
+        ...(runOptions.verbose ? [`--verbose`] : []),
+      ],
       {
-        project: remote,
-        target,
-        configuration: context.target.configuration,
-        runOptions,
-      },
-      options.verbose
-    ).then((obs) => {
-      obs.toPromise().catch((err) => {
-        throw new Error(
-          `Remote '${remote}' failed to serve correctly due to the following: \r\n${err.toString()}`
-        );
-      });
+        cwd: context.workspaceRoot,
+        stdio: ['ignore', 'pipe', 'pipe', 'ipc'],
+      }
+    );
+
+    remoteProcess.stdout.on('data', (data) => {
+      if (isCollectingStaticRemoteOutput) {
+        outWithErr.push(data.toString());
+      } else {
+        outWithErr = null;
+        remoteProcess.stdout.removeAllListeners('data');
+      }
+    });
+    remoteProcess.stderr.on('data', (data) => logger.info(data.toString()));
+    remoteProcess.on('exit', (code) => {
+      if (code !== 0) {
+        logger.info(outWithErr.join(''));
+        throw new Error(`Remote failed to start. See above for errors.`);
+      }
     });
+    process.on('SIGTERM', () => remoteProcess.kill('SIGTERM'));
+    process.on('exit', () => remoteProcess.kill('SIGTERM'));
   }
 
-  return executeWebpackDevServerBuilder(options, context);
+  const waitForRemotes = async () => {
+    if (remotePorts.size === 0) {
+      return true;
+    }
+    try {
+      await Promise.all(
+        [...remotePorts.values()].map((port) =>
+          // Allow 20 minutes for each remote to start, which is plenty of time but we can tweak it later if needed.
+          // Most remotes should start in under 1 minute.
+          waitForPortOpen(port, {
+            retries: 480,
+            retryDelay: 2500,
+            host: 'localhost',
```

Comment:
This goes for the React version as well:

- Is there a potential issue if 'localhost' is started on `::1` (ipv6) but the server is `127.0.0.1` (ipv4)?
- If the project config sets the hostname to something else other than `localhost` this wouldn't work

Reply:
This isn't needed anymore thanks to fixing the TS issue by using `registerTsProject`